### PR TITLE
modify non-handler methods to not write a response

### PR DESCRIPTION
**Pull request by ewollesen**
In prepping the send invitation endpoint for care partner alerting, it has become necessary to have a new endpoint that won't error immediately when an invitee already has an accepted invitation from the invitor.

Since checking for duplicate invitations is firmly in the realm of business logic, this separation also makes sense, as keeping transport logic (HTTP-related things) separate from business logic makes for easier understanding and simpler tests too.

Part of BACK-2500

### CODE REVIEW

File: api/invite.go [hunk 1]
```diff
@@ -411,11 +406,18 @@ func (a *Api) SendInvite(res http.ResponseWriter, req *http.Request, vars map[st
...
 			log.Printf("SendInvite: invited [%s] user already has or had an invite", ib.Email)
+			statusErr := &status.StatusError{
+				Status: status.NewStatus(http.StatusConflict, statusExistingInviteMessage),
+			}
+			a.sendModelAsResWithStatus(res, statusErr, http.StatusConflict)
 			return
-		} else if alreadyMember, invitedUsr := a.checkAccountAlreadySharedWithUser(invitorID, ib.Email, res); alreadyMember {
+		} else if alreadyMember, invitedUsr := a.checkAccountAlreadySharedWithUser(invitorID, ib.Email); alreadyMember {
 			a.logger.Infof("invited [%s] user is already a member of the care team of %v", ib.Email, invitorID)
+			log.Println(statusExistingMemberMessage)
```
Comment (tjotala) [inline api/invite.go:418]: Just curious: any reason to bring this `log.Println` to the top-level code instead of leaving in `checkAccountAlready...`

**Review by tjotala — APPROVED** [review]
LGTM, just curiosity question

File: api/invite.go (hunk 1 above)
Comment (ewollesen) [reply api/invite.go:418]: Nope, overzealous cut & paste. I'll fix really quick.

**Comment by ewollesen** [comment]
Heh, had to force push to prevent that ugly merge commit. :D

**Review by tjotala — APPROVED** [review]
LGTM, still :)

**Pull request merged**